app/views/events/index.html.haml

```diff
@@ -1,6 +1,12 @@
 - events.each do |event|
   .event
-    %h1
```

Comment:
You could keep the h1 top level (instead of duplicating the h1 tag in each conditional) and nest the logic one level farther in and it should work well. You could then put the small after the conditional logic as well since its the same for each.